### PR TITLE
Multiple dose compartment

### PR DESCRIPTION
**Pull request by JohanWBorg**
Changed CompartmentalSystem.dosing_compartment to tuple allowing for multiple dose compartments

### CODE REVIEW

File: src/pharmpy/model/statements.py [hunk 1]
```diff
@@ -908,24 +908,32 @@ def get_n_connected(self, comp):
...
-        >>> model.statements.ode_system.dosing_compartment
+        >>> model.statements.ode_system.dosing_compartment[0]
         Compartment(CENTRAL, amount=A_CENTRAL, dose=Bolus(AMT, admid=1))
         """
+        dosing_comps = tuple()
         for node in _comps(self._g):
             if node.dose is not None:
-                return node
+                if node.name != self.central_compartment.name:
+                    dosing_comps = tuple([node]) + dosing_comps
```
Comment (stellabelin) [inline src/pharmpy/model/statements.py:930]: How come it is `tuple([node]) + dosing_comps`, isn't it possible to do `(node,) + dosing_comps`?

File: src/pharmpy/model/statements.py [hunk 2]
```diff
@@ -908,24 +908,32 @@ def get_n_connected(self, comp):
...
+                else:
+                    dosing_comps = dosing_comps + tuple([node])
```
Comment (stellabelin) [inline src/pharmpy/model/statements.py:932]: Same as above

File: tests/model/test_statements.py [hunk 3]
```diff
@@ -478,3 +478,17 @@ def test_assignment_create_numeric(load_model_for_test, testdata):
...
+    model = load_model_for_test(testdata / 'nonmem' / 'pheno.mod')
+    model = set_first_order_absorption(model)
+
+    ode = model.statements.ode_system
+    cb = CompartmentalSystemBuilder(ode)
+    cb.set_dose(cb.find_compartment("CENTRAL"), cb.find_compartment("DEPOT").dose)
+    ode = CompartmentalSystem(cb)
+    model = model.replace(
+        statements=model.statements.before_odes + ode + model.statements.after_odes
+    )
```
Comment (stellabelin) [inline tests/model/test_statements.py:493]: Very small thing but how come this is in the test? The `assert` doesn't use the model object?

File: src/pharmpy/model/statements.py [hunk 4]
```diff
@@ -908,24 +908,32 @@ def get_n_connected(self, comp):
     def dosing_compartment(self):
         """The dosing compartment
 
-        A dosing compartment is a compartment that receives an input dose. Only one dose
-        compartment is supported.
+        A dosing compartment is a compartment that receives an input dose. Multiple
+        dose compartments are supported.
```
Comment (stellabelin) [inline src/pharmpy/model/statements.py:912]: Maybe it would be good here to write that we define the dose into central as the last dose compartment?

**Review by stellabelin — COMMENTED** [review]
Looks good, some minor comments/questions.

File: src/pharmpy/model/statements.py (hunk 2 above)
Comment (JohanWBorg) [reply src/pharmpy/model/statements.py:932]: See above

File: tests/model/test_statements.py (hunk 3 above)
Comment (JohanWBorg) [reply tests/model/test_statements.py:493]: It only served a purpose when checking the model myself. It has now been removed!

File: src/pharmpy/model/statements.py (hunk 4 above)
Comment (JohanWBorg) [reply src/pharmpy/model/statements.py:912]: Added description :)

**Review by stellabelin — APPROVED** [review]
LGTM!

**Pull request merged**